src/oemof/solph/components.py

```diff
@@ -246,6 +249,13 @@ def _check_invest_attributes(self):
 
         self._set_flows()
 
+    def _check_number_of_flows(self):
+        msg = "Only one {0} flow allowed in the GenericStorage {1}."
+        if len(self.inputs) > 1:
+            raise AttributeError(msg.format("input", self.label))
+        if len(self.outputs) > 1:
```

Comment:
Should this be `== 1` instead of `> 1` (in both checks)? I haven't checked, but I would expect that without inputs or outputs, the code does the equivalent of `list()[0]`, causing an exception to be thrown.

Reply:
You cannot do it because it is allowed to initialise a GenericStorage without inputs and outputs even though it is unwanted in most situation. For any reason users may want to add these attributes later.

I added a `SuspiciousUsageWarning` which users can switch off if they know what they are doing. This warning is raised if no inputs or outputs are defined (including misspelling).